### DISCUSSION TITLE
What is the Interlisp Way?

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#1095] [General]
Interlisp wasn't just an IDE, there was an Interlisp Way.

Here is an example:
@pamoroso asks on Libera Chat Interlisp channel:

> Suppose the buffer of a TEdit window has been modified but not saved. Is quitting via titlebar middle-click Quit or right-click Close supposed to prompt whether to save the buffer?

@masinter first replied: 
> i don't know... the Get Put menu for opening and saving files always seemed awkward to me

but later answered:

> I don't remember seeing a "do you want to save before I quit" behavior... it's not part of the Interlisp Way
> The Interlisp Way is "save early and often"

**Comment by lassik** [comment]
[Obligatory link to [The Zen of Python](https://peps.python.org/pep-0020/)]

**Comment by rmkaplan** [comment]
For me, there is a * in the window to indicate that it has been modified.  Attempting to close it brings up a question in the prompt window, asking if I really mean it.

Do you not see that behavior?

**Comment by pamoroso** [comment]
Thanks, I wondered because of what Steve Kaisler wrote in section *1.1.7.4 Exiting TEdit* on page 26 of his book [Medley Interlisp: Interactive Programming Tools](https://interlisp.org/documentation/2021-interlisp-book-3.pdf):

> Selecting the Quit command from the TEdit command menu allows you to exit TEdit without saving the file that you are editing. If you have made changes to the file since you last saved it on the storage media, you will be asked to confirm the immediate exit.

Although `Quit` does prompt to save an unsaved buffer, `Close` does not. And yes, I do see the `*` modified buffer marker.

A Zen of Interlisp is a great idea. Another entry may be "Embrace the mouse".

**Comment by masinter** [comment]
I was thinking DWIM was a kind of philosophy as well: that the environment should go ahead and coerce things without complaint. So symbol -> string -> pathname -> .... 

This is in support of the idea that declarations and unsafe optimizations weren't important for the most part.
